dev/release/verify-release-candidate.sh

```diff
@@ -282,6 +282,40 @@ ${ARROW_CMAKE_OPTIONS}
   popd
 }
 
+test_csharp() {
+  pushd csharp
+
+  if ! which dotnet > /dev/null 2>&1; then
+    if [ "$(uname)" == "Darwin" ]; then
+      DOTNET_URL=https://download.visualstudio.microsoft.com/download/pr/328d95ad-eb94-4d1f-a4a9-2a35dbfea719/4913918a65ae135af1c8823f04061708/dotnet-sdk-2.2.300-osx-x64.tar.gz
+    else
+      DOTNET_URL=https://download.visualstudio.microsoft.com/download/pr/5e92f45b-384e-41b9-bf8d-c949684e20a1/67a98aa2a4e441245d6afe194bd79b9b/dotnet-sdk-2.2.300-linux-x64.tar.gz
```

Comment:
Should we use  https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-2.2.300-linux-x64-binaries?
But I didn't know how to install *.tar.gz.
(I tried `curl -O -L https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-2.2.300-linux-x64-binaries`)

Reply:
The page uses JavaScript (`window.open("...")`) to download `tar.gz` automatically.
So we need to extract URL from HTML:

```shell
curl https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-2.2.300-linux-x64-binaries | \
  grep 'window\.open' | \
  grep -E -o '[^"]+' | \
  sed -n '2p'
# https://download.visualstudio.microsoft.com/download/pr/5e92f45b-384e-41b9-bf8d-c949684e20a1/67a98aa2a4e441245d6afe194bd79b9b/dotnet-sdk-2.2.300-linux-x64.tar.gz
```

Reply:
I see. Thanks for your comment.